test/OpenTelemetry.Instrumentation.AspNetCore.Tests/IncomingRequestsCollectionsIsAccordingToTheSpecTests_Dupe.cs

```diff
@@ -0,0 +1,204 @@
+// <copyright file="IncomingRequestsCollectionsIsAccordingToTheSpecTests_Dupe.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Diagnostics;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using OpenTelemetry.Trace;
+using TestApp.AspNetCore;
+using Xunit;
+
+namespace OpenTelemetry.Instrumentation.AspNetCore.Tests
+{
+    public class IncomingRequestsCollectionsIsAccordingToTheSpecTests_Dupe
+        : IClassFixture<WebApplicationFactory<Program>>
+    {
+        private readonly WebApplicationFactory<Program> factory;
+
+        public IncomingRequestsCollectionsIsAccordingToTheSpecTests_Dupe(WebApplicationFactory<Program> factory)
+        {
+            this.factory = factory;
+        }
+
+        [Theory]
+        [InlineData("/api/values", null, "user-agent", 503, "503")]
+        [InlineData("/api/values", "?query=1", null, 503, null)]
+        [InlineData("/api/exception", null, null, 503, null)]
+        [InlineData("/api/exception", null, null, 503, null, true)]
+        public async Task SuccessfulTemplateControllerCallGeneratesASpan_Dupe(
```

Comment:
Could we move these two new tests to the existing `IncomingRequestsCollectionsIsAccordingToTheSpecTests` class? That way they could also reuse `ValidateTagValue` and `TestCallbackMiddlewareImpl` from the existing class.

Reply:
I split these into separate classes because I was getting concurrency issues causing test failures.
Also having it this way will make it easier to clean up when this library goes stable (ie: delete the old & dupe).